src/tests/test_magic.py

```diff
@@ -258,49 +261,69 @@ def test_persist_replace_no_override(ip, test_table, expected_result):
 
 
 @pytest.mark.parametrize(
-    "test_table, expected_result",
+    "first_test_table, second_test_table, expected_result",
     [
-        ("test", [(0, 1, "foo")]),
-        ("author", [(0, "William", "Shakespeare", 1616)]),
-        (
-            "website",
-            [
-                (
-                    0,
-                    "Bertold Brecht",
-                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
-                    1954,
-                )
-            ],
-        ),
-        ("number_table", [(0, 4, -2)]),
+        ("test", "author", [(0, "William", "Shakespeare", 1616)]),
+        ("author", "test", [(0, 1, "foo")]),
+        ("test", "number_table", [(0, 4, -2)]),
+        ("number_table", "test", [(0, 1, "foo")]),
     ],
 )
-def test_persist_replace_override(ip, test_table, expected_result):
-    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
+def test_persist_replace_override(
+    ip, first_test_table, second_test_table, expected_result
+):
+    saved_df_name = "dummy_df_name"
+    get_table_rows_as_dataframe(
+        ip, table_name=first_test_table, desire_saved_df_name=saved_df_name
+    )
     ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
-    saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
-    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+    get_table_rows_as_dataframe(
```

Comment:
To make it more readable, let's use the returned value

```
table_df = get_table_rows_as_dataframe(ip, table)
```

(this is relevant for other places as well)